subsys/bluetooth/audio/mcc.c

```diff
@@ -1193,21 +1180,290 @@ static uint8_t discover_include_func(struct bt_conn *conn,
 	}
 	return BT_GATT_ITER_STOP;
 }
+
+/* Start discovery of included services */
+static void discover_included(struct bt_conn *conn)
+{
+	int err;
+
+	discover_params.start_handle = cur_mcs_inst->start_handle;
+	discover_params.end_handle = cur_mcs_inst->end_handle;
+	discover_params.type = BT_GATT_DISCOVER_INCLUDE;
+	discover_params.func = discover_include_func;
+
+	BT_DBG("Start discovery of included services");
+	err = bt_gatt_discover(conn, &discover_params);
+	if (err) {
+		BT_DBG("Discover of included service failed");
+		discovery_failure(conn, err);
+	}
+}
 #endif /* CONFIG_BT_MCC_OTS */
 
+static void subscribe_mcs_char_func(struct bt_conn *conn);
 
-/* This function is called when characteristics are found.
- * The function will store handles, and optionally subscribe to, GMCS
- * characteristics.
+static void subscription_cb(struct bt_conn *conn, uint8_t err,
+			    struct bt_gatt_subscribe_params *params)
+{
+	if (err) {
+		BT_DBG("Subscription callback error");
+		discovery_failure(conn, err);
+		return;
+	}
+
+	BT_DBG("Subscribed: value handle: %d, ccc handle: %d",
+	       params->value_handle, params->ccc_handle);
+
+	/* Subscribe to next characteristic */
+	subscribe_mcs_char_func(conn);
+}
+
+/* Subscribe to a characteristic - helper function */
+static int do_subscribe(struct bt_conn *conn, uint16_t handle,
+			 struct bt_gatt_subscribe_params *sub_params)
+{
+	/* With ccc_handle == 0 it will use auto discovery */
+	sub_params->ccc_handle = 0;
+	sub_params->end_handle = cur_mcs_inst->end_handle;
+	sub_params->value = BT_GATT_CCC_NOTIFY;
+	sub_params->value_handle = handle;
+	sub_params->notify = mcs_notify_handler;
+	sub_params->subscribe = subscription_cb;
+	/* disc_params pointer is also used as subscription flag */
+	sub_params->disc_params = &discover_params;
+
+	BT_DBG("Subscring to handle %d", handle);
+	return bt_gatt_subscribe(conn, sub_params);
+}
+
+/* This function will subscribe to GMCS CCCDs.
  * After this, the function will start discovery of included services.
  */
+static void subscribe_mcs_char_func(struct bt_conn *conn)
+{
+	struct bt_gatt_subscribe_params *sub_params = NULL;
+	int err = 0;
+
+	/* The characteristics may be in any order on the server, and
+	 * not all of them may exist => need to check all.
+	 * For each of the subscribable characteristics
+	 * - check if we have a handle for it
+	 * - check sub_params.disc_params pointer to see if we have
+	 *   already subscribed to it (set in do_subscribe() ).
+	 */
+
+	if (cur_mcs_inst->player_name_handle &&
+	    cur_mcs_inst->player_name_sub_params.disc_params == NULL) {
+		sub_params = &cur_mcs_inst->player_name_sub_params;
+		err = do_subscribe(conn, cur_mcs_inst->player_name_handle, sub_params);
+		if (err) {
+			BT_DBG("Could not subscribe");
```

Comment:
```suggestion
			BT_DBG("Could not subscribe: %d", err);
```

Ditto for the other places

Reply:
Fixed everywhere, I believe.